src/ol/render/canvas/TextBuilder.js

```diff
@@ -336,6 +339,7 @@ class CanvasTextBuilder extends CanvasBuilder {
         );
         if (textState.backgroundFill) {
           this.updateFillStyle(this.state, this.createFill);
+          this.state.fillStyle = defaultFillStyle;
```

Comment:
This looks scary, and unrelated to the pull request. Can you explain the rationale behind this added line?

Reply:
`backgroundFill` does not have a key for either rendering or hit detection, it uses the fillStyle directly, so that is needed to override transparent so transparent can be hit.

Reply:
This brings state and instructions out of sync. Did you mean to use `state.currentFillStyle`?

Reply:
Anyway, this does not look right. Text background is unfortunately not so well tested (my bad), and I'd like to avoid side effects. Better not implement hit-detection for transparent background fill, than risking a regression.

Reply:
To do this properly, 3 hit detection instructions would be needed:
* SET_FILL_STYLE with a non-transparent fill
* DRAW_IMAGE
* SET_FILL_STYLE restoring the previous fill style

Reply:
Is the restore best done before or after the END_GEOMETRY ?

Reply:
Do it right after DRAW_IMAGE.